open-api/rest-catalog-open-api.yaml

```diff
@@ -3129,6 +3145,11 @@ components:
          - `s3.secret-access-key`: secret for credentials that provide access to data in S3 
          - `s3.session-token`: if present, this value should be used for as the session token 
          - `s3.remote-signing-enabled`: if `true` remote signing should be performed as described in the `s3-signer-open-api.yaml` specification
+
+        ## Storage Credentials
+
+        Credentials for ADLS / GCS / S3 / ... are provided through the `storage-credentials` field.
```

Comment:
the documentation of the respective FileIO properties is handled outside this PR and is in the scope of https://github.com/apache/iceberg/pull/10576. We'll be updating/rebasing https://github.com/apache/iceberg/pull/10576 once this PR here is in and that should make it clearer for servers which properties are available for FileIO implementations

Reply:
thx!

Reply:
@dimas-b does the documentation of the storage related properties address your comment about "additional guidelines"?

Reply:
#10576 appears to define credential properties in the REST spec, which is fine from my POV :) I assume after rebasing it will be talking about credential properties under the new config section introduced in this PR.

Reply:
yes exactly, we're trying to move the docs to this new credential section